game/assets/scripts/dialogue/dialogue_player.lua

```diff
@@ -1,76 +1,170 @@
-local narrator = require("narrator/narrator")
-local story
-local ui
-local currentChoice = 1
-local maxChoices = 1
+DialoguePlayer = class("DialoguePlayer")
 
-exports = {
-	story = ""
-}
+function DialoguePlayer:initialize(entity)
+    self.exports = {
+		story = "",
+		up = "Up",
+		down = "Down",
+		next = "Next"
+	}
+end
+
+function DialoguePlayer:begin(entity)
+	self.narrator = require("narrator/narrator")
+	local book = self.narrator.parseFile(self.exports.story)
+	self.story = self.narrator.initStory(book)
+	self.story:begin()
+	self.ui = entity:getUI()
+	self.currentChoice = 1
+	self.maxChoices = 1
+	self.currentActor = ""
+	if self.story.variables["default_color"] then
+		self.default_color = self.story.variables["default_color"]
```

Comment:
camelCase variables only, so this means `self.defaultColor`

Reply:
done